pyiron_base/project/maintenance.py

```diff
@@ -60,6 +67,42 @@ def get_repository_status():
         return report
 
 
+class UpdateMaintenance:
+    def __init__(self, project):
+        self._project = project
+
+    def base_v0_3_to_v0_4(self, project=None):
+        """Update hdf files written with pyiron_base-0.3.x to pyiron_base-0.4.x
+
+        pyiron_base<=0.3.9 has a bug that writes all arrays with dtype=object even
+        numeric ones.  As a fix pyiron_base=0.4.0 introduces a conversion when reading
+        such arrays, but does not automatically save them.  This conversion script
+        simply goes over all jobs and rewrites their HDF5 files, since it's read with
+        the correct dtype, this then writes this correct dtype.
+
+        Args:
+            project(None/pyiron_project/"all"): The project to be converted from 0.3 to 0.4 ; default: current project
+                if "all" is provided, pyiron tries to find all projects using the PROJECT_PATHS defined in the
+                configuration.
+        """
+        if project is None:
```

Comment:
I like the option that the user can pass a list of projects or even choose all to be updated, but a clearer docstring, which explains different options, is missing.

Reply:
Have you changed the docstring, or I missed the part above the Args. If it is the later, sorry for this nonsense comment.

Reply:
Well, the part above the Args was there already. After all I improved the docstring to list all possibilities to define a project :)